src/App.tsx

```diff
@@ -1,26 +1,290 @@
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      })
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  useEffect(() => {
+    if (!errorMessage) {
+      return;
+    }
+
+    const timeout = setTimeout(() => setErrorMessage(''), 3000);
+
+    return () => clearTimeout(timeout);
+  }, [errorMessage]);
+
+  const handleDelete = (id: number) => {
+    setLoadingIds([id]);
+
+    deleteTodo(id)
+      .then(() => {
+        setTodos(currentTodo => currentTodo.filter(todo => todo.id !== id));
+      })
+      .catch(() => setErrorMessage('Unable to delete a todo'))
+      .finally(() => {
+        setLoadingIds([]);
+        inputRef.current?.focus();
+      });
+  };
+
+  const handleCompltete = (id: number) => {
+    const completedTodo: Todo[] = todos.map(todo => {
+      if (todo.id === id) {
+        setLoadingIds([...loadingIds, todo.id]);
+
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+
+      setTimeout(() => {
+        const loading = loadingIds.filter(item => {
+          return item !== todo.id;
+        });
+
+        setLoadingIds(loading);
+      }, 500);
+
+      return todo;
+    });
+
+    setTodos(completedTodo);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const trimmedTitle = title.trim();
+
+    if (trimmedTitle.length === 0) {
+      setErrorMessage('Title should not be empty');
+
+      return;
+    }
+
+    const newTodo = {
+      title: trimmedTitle,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    const tempTodo: Todo = {
+      id: 0,
+      title: trimmedTitle,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempState(tempTodo);
+
+    if (inputRef.current) {
+      inputRef.current.disabled = true;
+    }
+
+    postTodo(newTodo)
+      .then((newTodoFromServer: Todo) => {
+        setTodos(currentTodo => [...currentTodo, newTodoFromServer]);
+        setTitle('');
+      })
+      .catch(() => setErrorMessage('Unable to add a todo'))
+      .finally(() => {
+        if (inputRef.current) {
+          inputRef.current.disabled = false;
+        }
+
+        inputRef.current?.focus();
+
+        setTempState(null);
+      });
+  };
+
+  const activeTodo = [...todos].filter(todo => todo.completed === false);
+  const completedTodo = [...todos].filter(todo => todo.completed === true);
+
+  const handleDeleteCompleted = () => {
+    const deletedIds = completedTodo.map(todo => todo.id);
+
+    deletedIds.forEach(id => handleDelete(id));
+  };
+
+  const sortTodos = () => {
+    const filteredTodos = [...todos].filter(todo => {
+      if (sortBy === 'Active') {
```

Comment:
All statuses need to be in the enum, then you can use it In all places in your application.